File: app/src/main/java/com/example/fixitnopanic/BaseActivity.java
package com.example.fixitnopanic;

import android.os.Bundle;
import androidx.appcompat.app.AppCompatActivity;

public abstract class BaseActivity extends AppCompatActivity {
    protected SharedPrefsHelper sharedPrefsHelper;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        sharedPrefsHelper = new SharedPrefsHelper(this);
    }

    @Override
    public void onBackPressed() {
        // Всегда возвращаемся на главный экран при нажатии "Назад"
        sharedPrefsHelper.setCurrentScreen("main");
        super.onBackPressed();
    }
}

File: app/src/main/java/com/example/fixitnopanic/BaseActivity.kt
package com.example.fixitnopanic

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity

abstract class BaseActivity : AppCompatActivity() {
    protected lateinit var sharedPrefsHelper: SharedPrefsHelper

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        sharedPrefsHelper = SharedPrefsHelper(this)
    }

    override fun onBackPressed() {
        sharedPrefsHelper.setCurrentScreen("main")
        super.onBackPressed()
    }
}